experimental/zerotrie/tests/derive_test.rs

```diff
@@ -0,0 +1,85 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#![allow(non_camel_case_types, non_snake_case)]
+
+use zerotrie::ZeroTrie;
+use zerotrie::ZeroTrieExtendedCapacity;
+use zerotrie::ZeroTriePerfectHash;
+use zerotrie::ZeroTrieSimpleAscii;
+use zerovec::ZeroVec;
+
+#[cfg_attr(feature = "yoke", derive(yoke::Yokeable))]
+#[cfg_attr(feature = "zerofrom", derive(zerofrom::ZeroFrom))]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+#[cfg_attr(feature = "databake", derive(databake::Bake), databake(path = crate))]
+struct DeriveTest_ZeroTrie_ZeroVec<'data> {
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    _data: ZeroTrie<ZeroVec<'data, u8>>,
+}
+
+#[test]
+#[cfg(all(feature = "databake", feature = "alloc"))]
+fn bake_ZeroTrie_ZeroVec() {
+    use databake::*;
+    extern crate std;
+    test_bake!(
+        DeriveTest_ZeroTrie_ZeroVec<'static>,
+        crate::DeriveTest_ZeroTrie_ZeroVec {
```

Comment:
Please note that these tests are directly based on the ones in utils/zerovec/src/yoke_impls.rs

> wrapping the zerotrie in a struct doesn't add anything to the test

It tests that the zerotrie can be used in a struct with derives. Without this test, it's possible that the imps are missing or incomplete. This was a problem in the past which is why we added these tests in the zerovec crate. I think we should err on the side of testing this behavior for any zero-copy data structure.

> and it complicates it because it now also requires the struct's bake to be stable

Why? If the bake changes, update the test.

> I'd also prefer this as a unit test next to the code under test.

This doesn't catch problems like private fields. Given that this tests external behavior it seems better for this to be in an integration test.

Reply:
I agree with Shane here, this is our standard pattern for testing derive bounds for generic stuff